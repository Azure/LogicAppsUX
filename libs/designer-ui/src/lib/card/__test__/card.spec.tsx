import type { CardProps } from '../index';
import { Card } from '../index';
import React from 'react';
import renderer from 'react-test-renderer';
import { describe, vi, beforeEach, afterEach, beforeAll, afterAll, it, test, expect } from 'vitest';

describe('lib/card', () => {
  let minimal: CardProps;

  beforeEach(() => {
    minimal = {
      brandColor: '#474747',
      drag: () => null,
      dragPreview: () => null,
      draggable: false,
      id: 'id',
      title: 'title',
      isSecureInputsOutputs: false,
      runData: undefined,
    };
  });

  it('should render', () => {
    const tree = renderer.create(<Card {...minimal} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render as inactive', () => {
    const tree = renderer.create(<Card {...minimal} active={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render as cloned', () => {
    const tree = renderer.create(<Card {...minimal} cloned={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render as draggable', () => {
    const tree = renderer.create(<Card {...minimal} draggable={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render as selected', () => {
    const tree = renderer.create(<Card {...minimal} selectionMode="selected" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render as pinned', () => {
    const tree = renderer.create(<Card {...minimal} selectionMode="pinned" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it.each<[string, string | undefined, string | undefined]>([
    ['a connector name', 'SharePoint', undefined],
    ['an operation name', undefined, 'Get an item'],
    ['an operation name and connector name', 'SharePoint', 'Get an item'],
  ])('should render with an icon and %s (%#)', (_caseName, connectorName, operationName) => {
    const tree = renderer
      .create(
        <Card
          {...minimal}
          connectorName={connectorName}
          icon="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          operationName={operationName}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
