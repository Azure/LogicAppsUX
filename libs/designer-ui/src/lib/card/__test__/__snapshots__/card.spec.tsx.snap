// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`lib/card > should render 1`] = `
<div
  style={
    {
      "position": "relative",
    }
  }
>
  <div
    className="msla-panel-card-container"
    data-automation-id="card-title"
    data-testid="card-title"
    onClick={[Function]}
    onContextMenu={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    style={
      {
        "borderLeft": "4px solid rgba(71,71,71, 1)",
        "borderRadius": "2px",
      }
    }
    tabIndex={2}
  >
    <div
      className="msla-selection-box"
    />
    <div
      className="panel-card-main"
    >
      <div
        aria-label=" operation"
        className="panel-card-header"
        role="button"
      >
        <div
          className="panel-card-content-container"
        >
          <div
            className="panel-card-content-gripper-section"
          />
          <div
            className="panel-card-content-icon-section"
          >
            <div
              className="fui-Spinner r1k3z50n msla-card-header-spinner"
              role="progressbar"
            >
              <span
                className="fui-Spinner__spinner rvgcg50 ___ruqntd0_k4fqbj0 fjamq6b f64fuq3 f1v3ph3m"
              >
                <span
                  className="fui-Spinner__spinnerTail r1lregi5"
                />
              </span>
            </div>
          </div>
          <div
            className="panel-card-top-content"
          >
            <div
              className="panel-msla-title"
            >
              title
            </div>
          </div>
        </div>
      </div>
      <div
        className="msla-card-v2-footer"
      >
        <div
          className="msla-badges"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`lib/card > should render as cloned 1`] = `
<div
  style={
    {
      "position": "relative",
    }
  }
>
  <div
    className="msla-panel-card-container msla-card-ghost-image"
    data-automation-id="card-title"
    data-testid="card-title"
    onClick={[Function]}
    onContextMenu={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    style={
      {
        "borderLeft": "4px solid rgba(71,71,71, 1)",
        "borderRadius": "2px",
      }
    }
    tabIndex={2}
  >
    <div
      className="msla-selection-box"
    />
    <div
      className="panel-card-main"
    >
      <div
        aria-label=" operation"
        className="panel-card-header"
        role="button"
      >
        <div
          className="panel-card-content-container"
        >
          <div
            className="panel-card-content-gripper-section"
          />
          <div
            className="panel-card-content-icon-section"
          >
            <div
              className="fui-Spinner r1k3z50n msla-card-header-spinner"
              role="progressbar"
            >
              <span
                className="fui-Spinner__spinner rvgcg50 ___ruqntd0_k4fqbj0 fjamq6b f64fuq3 f1v3ph3m"
              >
                <span
                  className="fui-Spinner__spinnerTail r1lregi5"
                />
              </span>
            </div>
          </div>
          <div
            className="panel-card-top-content"
          >
            <div
              className="panel-msla-title"
            >
              title
            </div>
          </div>
        </div>
      </div>
      <div
        className="msla-card-v2-footer"
      >
        <div
          className="msla-badges"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`lib/card > should render as draggable 1`] = `
<div
  style={
    {
      "position": "relative",
    }
  }
>
  <div
    className="msla-panel-card-container"
    data-automation-id="card-title"
    data-testid="card-title"
    onClick={[Function]}
    onContextMenu={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    style={
      {
        "borderLeft": "4px solid rgba(71,71,71, 1)",
        "borderRadius": "2px",
      }
    }
    tabIndex={2}
  >
    <div
      className="msla-selection-box"
    />
    <div
      className="panel-card-main"
    >
      <div
        aria-label=" operation"
        className="panel-card-header"
        role="button"
      >
        <div
          className="panel-card-content-container"
        >
          <div
            className="panel-card-content-gripper-section draggable"
          >
            <svg
              fill="none"
              height="12"
              viewBox="0 0 2 10"
              width="4"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 2V0H2V2H0ZM0 6V4H2V6H0ZM0 10V8H2V10H0Z"
                fill="#605E5C"
              />
            </svg>
          </div>
          <div
            className="panel-card-content-icon-section"
          >
            <div
              className="fui-Spinner r1k3z50n msla-card-header-spinner"
              role="progressbar"
            >
              <span
                className="fui-Spinner__spinner rvgcg50 ___ruqntd0_k4fqbj0 fjamq6b f64fuq3 f1v3ph3m"
              >
                <span
                  className="fui-Spinner__spinnerTail r1lregi5"
                />
              </span>
            </div>
          </div>
          <div
            className="panel-card-top-content"
          >
            <div
              className="panel-msla-title"
            >
              title
            </div>
          </div>
        </div>
      </div>
      <div
        className="msla-card-v2-footer"
      >
        <div
          className="msla-badges"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`lib/card > should render as inactive 1`] = `
<div
  style={
    {
      "position": "relative",
    }
  }
>
  <div
    className="msla-panel-card-container inactive"
    data-automation-id="card-title"
    data-testid="card-title"
    onClick={[Function]}
    onContextMenu={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    style={
      {
        "borderLeft": "4px solid rgba(71,71,71, 1)",
        "borderRadius": "2px",
      }
    }
    tabIndex={2}
  >
    <div
      className="msla-selection-box"
    />
    <div
      className="panel-card-main"
    >
      <div
        aria-label=" operation"
        className="panel-card-header"
        role="button"
      >
        <div
          className="panel-card-content-container"
        >
          <div
            className="panel-card-content-gripper-section"
          />
          <div
            className="panel-card-content-icon-section"
          >
            <div
              className="fui-Spinner r1k3z50n msla-card-header-spinner"
              role="progressbar"
            >
              <span
                className="fui-Spinner__spinner rvgcg50 ___ruqntd0_k4fqbj0 fjamq6b f64fuq3 f1v3ph3m"
              >
                <span
                  className="fui-Spinner__spinnerTail r1lregi5"
                />
              </span>
            </div>
          </div>
          <div
            className="panel-card-top-content"
          >
            <div
              className="panel-msla-title"
            >
              title
            </div>
          </div>
        </div>
      </div>
      <div
        className="msla-card-v2-footer"
      >
        <div
          className="msla-badges"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`lib/card > should render as selected 1`] = `
<div
  style={
    {
      "position": "relative",
    }
  }
>
  <div
    className="msla-panel-card-container msla-panel-card-container-selected"
    data-automation-id="card-title"
    data-testid="card-title"
    onClick={[Function]}
    onContextMenu={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    style={
      {
        "borderLeft": "4px solid rgba(71,71,71, 1)",
        "borderRadius": "2px",
      }
    }
    tabIndex={2}
  >
    <div
      className="msla-selection-box selected"
    />
    <div
      className="panel-card-main"
    >
      <div
        aria-label=" operation"
        className="panel-card-header"
        role="button"
      >
        <div
          className="panel-card-content-container"
        >
          <div
            className="panel-card-content-gripper-section"
          />
          <div
            className="panel-card-content-icon-section"
          >
            <div
              className="fui-Spinner r1k3z50n msla-card-header-spinner"
              role="progressbar"
            >
              <span
                className="fui-Spinner__spinner rvgcg50 ___ruqntd0_k4fqbj0 fjamq6b f64fuq3 f1v3ph3m"
              >
                <span
                  className="fui-Spinner__spinnerTail r1lregi5"
                />
              </span>
            </div>
          </div>
          <div
            className="panel-card-top-content"
          >
            <div
              className="panel-msla-title"
            >
              title
            </div>
          </div>
        </div>
      </div>
      <div
        className="msla-card-v2-footer"
      >
        <div
          className="msla-badges"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`lib/card > should render with an icon and a connector name (0) 1`] = `
<div
  style={
    {
      "position": "relative",
    }
  }
>
  <div
    className="msla-panel-card-container"
    data-automation-id="card-title"
    data-testid="card-title"
    onClick={[Function]}
    onContextMenu={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    style={
      {
        "borderLeft": "4px solid rgba(71,71,71, 1)",
        "borderRadius": "2px",
      }
    }
    tabIndex={2}
  >
    <div
      className="msla-selection-box"
    />
    <div
      className="panel-card-main"
    >
      <div
        aria-label="SharePoint connector"
        className="panel-card-header"
        role="button"
      >
        <div
          className="panel-card-content-container"
        >
          <div
            className="panel-card-content-gripper-section"
          />
          <div
            className="panel-card-content-icon-section"
          >
            <img
              alt=""
              className="panel-card-icon"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            />
          </div>
          <div
            className="panel-card-top-content"
          >
            <div
              className="panel-msla-title"
            >
              title
            </div>
          </div>
        </div>
      </div>
      <div
        className="msla-card-v2-footer"
      >
        <div
          className="msla-badges"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`lib/card > should render with an icon and an operation name (1) 1`] = `
<div
  style={
    {
      "position": "relative",
    }
  }
>
  <div
    className="msla-panel-card-container"
    data-automation-id="card-title"
    data-testid="card-title"
    onClick={[Function]}
    onContextMenu={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    style={
      {
        "borderLeft": "4px solid rgba(71,71,71, 1)",
        "borderRadius": "2px",
      }
    }
    tabIndex={2}
  >
    <div
      className="msla-selection-box"
    />
    <div
      className="panel-card-main"
    >
      <div
        aria-label="Get an item operation"
        className="panel-card-header"
        role="button"
      >
        <div
          className="panel-card-content-container"
        >
          <div
            className="panel-card-content-gripper-section"
          />
          <div
            className="panel-card-content-icon-section"
          >
            <img
              alt=""
              className="panel-card-icon"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            />
          </div>
          <div
            className="panel-card-top-content"
          >
            <div
              className="panel-msla-title"
            >
              title
            </div>
          </div>
        </div>
      </div>
      <div
        className="msla-card-v2-footer"
      >
        <div
          className="msla-badges"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`lib/card > should render with an icon and an operation name and connector name (2) 1`] = `
<div
  style={
    {
      "position": "relative",
    }
  }
>
  <div
    className="msla-panel-card-container"
    data-automation-id="card-title"
    data-testid="card-title"
    onClick={[Function]}
    onContextMenu={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    style={
      {
        "borderLeft": "4px solid rgba(71,71,71, 1)",
        "borderRadius": "2px",
      }
    }
    tabIndex={2}
  >
    <div
      className="msla-selection-box"
    />
    <div
      className="panel-card-main"
    >
      <div
        aria-label="Get an item operation, SharePoint connector"
        className="panel-card-header"
        role="button"
      >
        <div
          className="panel-card-content-container"
        >
          <div
            className="panel-card-content-gripper-section"
          />
          <div
            className="panel-card-content-icon-section"
          >
            <img
              alt=""
              className="panel-card-icon"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            />
          </div>
          <div
            className="panel-card-top-content"
          >
            <div
              className="panel-msla-title"
            >
              title
            </div>
          </div>
        </div>
      </div>
      <div
        className="msla-card-v2-footer"
      >
        <div
          className="msla-badges"
        />
      </div>
    </div>
  </div>
</div>
`;
